### PR TITLE
[IN-63][Reviews] Use ember-osf queryHasMany

### PR DESCRIPTION
**Pull request by fabmiz**
## Purpose

Use ember-osf version of `queryHasMany`

## Summary of Changes

- Update usage of `queryHasMany` to `model.queryHasMany`  instead of `queryHasMany` from reviews `store`.
- Delete store
- Update tests

## Side Effects / Testing Notes

These changes affect the preprint detail page and the moderation list page.
On the preprint detail page:
- Make sure it works as usual with a simple preprint.
- Create a preprint with more than 10 contributors, and see if the preprint detail page works
with no errors (please check the console as well.)

On the moderation list page:
- Simply make sure it works as usual, no errors.

## Ticket

[IN-63](https://openscience.atlassian.net/browse/IN-63)

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "test:cover": "COVERAGE=true yarn test"
   },
   "devDependencies": {
-    "@centerforopenscience/ember-osf": "https://github.com/CenterForOpenScience/ember-osf#feature/reviews#2017-12-21",
+    "@centerforopenscience/ember-osf": "https://github.com/CenterForOpenScience/ember-osf#df7716240c43af4ef42a191985436279e6e510ec",
```
Comment (alexschiller) [inline package.json]: Can you run:
`yarn use-ember-osf-next-interfaces`
to set this to next interfaces instead of a pinned commit?

**Review by alexschiller — CHANGES_REQUESTED** [review]
swap to next interfaces for ember-osf otherwise looks fine

File: package.json (hunk 1 above)
Comment (fabmiz) [reply package.json]: Sure. it does not seem like `use-ember-osf-next-interfaces` script is defined here yet both develop and release/next-interfaces. I will get it from https://github.com/CenterForOpenScience/ember-osf-preprints/blob/develop/package.json#L19.

**Review by alexschiller — APPROVED** [review]

**Review by alexschiller — CHANGES_REQUESTED** [review]
CHANGELOG please

**Review by alexschiller — APPROVED** [review]

**Pull request merged**